internal/platform/utimes.go

```diff
@@ -3,33 +3,140 @@ package platform
 import (
 	"io/fs"
 	"syscall"
+	"time"
+	"unsafe"
 )
 
-// UtimesNano is like syscall.UtimesNano. This returns syscall.ENOENT if the
-// path doesn't exist.
+const (
+	// UTIME_NOW is a special syscall.Timespec NSec value used to set the
+	// file's timestamp to a value close to, but not greater than the current
+	// system time.
+	UTIME_NOW = -1
+
+	// UTIME_OMIT is a special syscall.Timespec NSec value used to avoid
+	// setting the file's timestamp.
+	UTIME_OMIT = -2
+)
+
+// Utimesns set file access and modification times on a path resolved to the
+// current working directory, at nanosecond precision.
 //
-// Note: This is like the function `utimensat` with `AT_FDCWD` in POSIX.
-// See https://pubs.opengroup.org/onlinepubs/9699919799/functions/futimens.html
-func UtimesNano(path string, atimeNsec, mtimeNsec int64) error {
-	err := syscall.UtimesNano(path, []syscall.Timespec{
-		syscall.NsecToTimespec(atimeNsec),
-		syscall.NsecToTimespec(mtimeNsec),
-	})
+// # Parameters
+//
+// The `times` parameter includes the access and modification timestamps to
+// assign. Special syscall.Timespec NSec values UTIME_NOW and UTIME_OMIT may be
+// specified instead of real timestamps. A nil `times` parameter behaves the
+// same as if both were set to UTIME_NOW.
+//
+// When the `symlinkFollow` parameter is true and the path is a symbolic link,
+// the target of expanding that link is updated.
+//
+// # Errors
+//
+// The following errors are expected:
+//   - syscall.EINVAL: `path` is invalid.
+//   - syscall.EEXIST: `path` exists and is a directory.
+//   - syscall.ENOTDIR: `path` exists and is a file.
+//
+// # Notes
+//
+//   - This is similar to syscall.UtimesNano, except that doesn't have flags to
+//     control expansion of symbolic links. It also doesn't support special
+//     values UTIME_NOW or UTIME_NOW.
+//   - This is like `utimensat` with `AT_FDCWD` in POSIX. See
+//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/futimens.html
+func Utimesns(path string, times *[2]syscall.Timespec, symlinkFollow bool) error {
```

Comment:
pointer signature allows us to tell nil from not. symlinkFollow is lifted from WASI in recognition there's only one real flag to handle